### DISCUSSION TITLE
XOutput and Optimization

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#303] [Ideas]
Hello,

I think there may be an issue with XOutput and optimizations. 
All the objectives as well as the datakeepers of an optimization are 'DataKeeper' and are stored in the xoutput_dict.

Now, the plot_pareto method of XOutput use all this datakeepers to compute the pareto front, not only the objectives. I don't know if this is makes any difference for the results (due to the nature of the pareto front), but I guess it may. (I haven't tested.)
That one can not distinguish objectives from normal datakeepers may also be a general issue, given that one can not access the optimization that computed the results.

So if you also think that could be an issue, my proposal would be to simply inherit an 'Objectiv' or 'ObjectivData' class from 'DataKeeper'. What do you think?

Best regards, Sebastian

**Comment by BonneelP** [comment]
Hello Sebastian,

Great idea :) Indeed we need to be able to sort the Datakeeper between Objectives and "normal" Datakeeper": it will be useful for the post-processings. 

Best regards,
Pierre

**Comment by SebGue** [comment]
Hello Pierre, 

do you mind if I change plot_pareto to show the original index instead of the 'is_valid' index? So it will be easier to to access these individuals.

Best, Sebastian

**Comment by BonneelP** [comment]
Yes, it will be more general that way I think.
Best regards,
Pierre
